pkg/controller/jobframework/reconciler.go

```diff
@@ -1148,22 +1150,72 @@ func (r *JobReconciler) ignoreUnretryableError(log logr.Logger, err error) error
 	return err
 }
 
-func generatePodsReadyCondition(job GenericJob, wl *kueue.Workload) metav1.Condition {
-	conditionStatus := metav1.ConditionFalse
-	message := "Not all pods are ready or succeeded"
-	// Once PodsReady=True it stays as long as the workload remains admitted to
-	// avoid unnecessary flickering the condition when the pods transition
-	// from Ready to Completed. As pods finish, they transition first into the
-	// uncountedTerminatedPods staging area, before passing to the
-	// succeeded/failed counters.
-	if workload.IsAdmitted(wl) && (job.PodsReady() || apimeta.IsStatusConditionTrue(wl.Status.Conditions, kueue.WorkloadPodsReady)) {
-		conditionStatus = metav1.ConditionTrue
-		message = "All pods were ready or succeeded since the workload admission"
+func generatePodsReadyCondition(log logr.Logger, job GenericJob, wl *kueue.Workload) metav1.Condition {
+	status := metav1.ConditionFalse
+	message := "Workload isn't admitted"
+	reason := kueue.WorkloadWaitForPodsStart
+	if !workload.IsAdmitted(wl) {
+		// The workload has not been admitted yet
+		// or it was admitted in the past but it's evicted/requeued
+		return metav1.Condition{
+			Type:    kueue.WorkloadPodsReady,
+			Status:  status,
+			Reason:  reason,
+			Message: message,
+			// ObservedGeneration is added via workload.UpdateStatus
+		}
+	}
+
+	podsReadyStartCond := workload.IsPodsReadyStartCond(wl)
+	podsReadyRecoveryCond := workload.IsPodsReadyRecoveryCond(wl)
+	log.V(2).Info("PATRYK before switch",
```

Comment:
Revisit the log. Surely remove PATRYK, but also probably this is V(3). I also think the ":" aren't necessary. Isn't the log folded as json?